File: opcall/inputs/validator_test.go
package inputs

import (
	"errors"
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
	"github.com/opspec-io/sdk-golang/model"
)

var _ = Describe("Validate", func() {
	objectUnderTest := newValidator()
	Context("invoked w/ nil param", func() {
		It("should return expected error", func() {
			/* arrange */
			providedValue := &model.Data{}

			expectedErrs := []error{errors.New("param required")}

			/* act */
			actualErrs := objectUnderTest.Validate(providedValue, nil)
			Expect(actualErrs).To(Equal(expectedErrs))
		})
	})
})

File: lib/op/interpreter/opcall/params/param/boolean/validator.test.js
const objectUnderTest = require('./validator')

describe('validate', () => {
  describe('typeof value isn\'t boolean', () => {
    it('returns expected result', () => {
      /* arrange */
      const expectedResult = [new Error('boolean required')]

      /* act */
      const actualResult = objectUnderTest.validate(undefined)

      /* assert */
      expect(actualResult).toEqual(expectedResult)
    })
  })
  describe('typeof value is boolean', () => {
    it('returns expected result', () => {
      /* arrange/act */
      const actualResult = objectUnderTest.validate(true)

      /* assert */
      expect(actualResult).toEqual([])
    })
  })
})
